test/integration/bind.py

```diff
@@ -0,0 +1,94 @@
+#!/usr/bin/env python3
+
+"""Test binding to interfaces and addresses."""
+
+import subprocess as subp
+import typing as T
+
+from testlib import check, util
+from testlib.log import log
+from testlib.test import Test
+from testlib.external import link_add
+
+util.require_root()
+util.require_command("ss", "-ntl")
+
+ADDRESSES = ("127.253.85.1", "127.253.85.2", "127.253.85.3")
+
+
+def get_listen_sockets(proto: str) -> str:
+    """Get a list of the currently listening sockets (ss output)."""
+    listen = subp.run(
+        ["ss", f"-nl{proto[0]}"], check=True, stdout=subp.PIPE, encoding="utf-8"
+    )
+    return listen.stdout
```

Comment:
Split into two tests as you suggested. One of them is still using `ss` because we need some way to check that UDP sockets are listening. Since we need to figure out which interfaces and addresses are available, and there's no way to do this in a cross-platform way, the test is tied to Linux and `ip addr` in any case.

TCP is checked with Python's `socket`.